### PR TITLE
Feature : printing block HR in npz format

### PR DESCRIPTION
**Pull request by wenfei-li**
See #3671

### CODE REVIEW

File: source/module_io/input.h [hunk 1]
```diff
@@ -271,7 +271,9 @@ class Input
     std::vector<int> out_band; // band calculation pengfei 2014-10-13
     bool out_proj_band; // projected band structure calculation jiyy add 2022-05-11
     std::vector<int> out_mat_hs; // output H matrix and S matrix in local basis.
-    bool out_mat_xc; // output exchange-correlation matrix in KS-orbital representation.
+    bool out_mat_xc; 
+    bool out_hr_npz;// output exchange-correlation matrix in KS-orbital representation.
```
Comment (dyzheng) [inline source/module_io/input.h:275]: Would you like to add documents for these new parameters?

**Comment by wenfei-li** [comment]
The fact that this pr remains open is quite amusing imo

File: source/module_io/input.h [hunk 2]
```diff
@@ -271,7 +271,10 @@ class Input
...
+    bool out_dm_npz;
+    bool dm_to_rho;
```
Comment (WHUweiqingzhou) [inline source/module_io/input.h:277]: It is quite hard to understand what `dm_to_rho` control, could you add some docs about it? Of course, add docs for other parameters, too.

File: source/module_io/input_conv.cpp [hunk 3]
```diff
@@ -685,6 +685,10 @@ void Input_Conv::Convert(void)
     hsolver::HSolverLCAO<std::complex<double>>::out_mat_t = INPUT.out_mat_t;
     hsolver::HSolverLCAO<std::complex<double>>::out_mat_dh = INPUT.out_mat_dh;
     GlobalV::out_mat_xc = INPUT.out_mat_xc;
+    GlobalV::out_hr_npz = INPUT.out_hr_npz;
```
Comment (WHUweiqingzhou) [inline source/module_io/input_conv.cpp:688]: please add UnitTests in `InputConvTest`

File: source/module_io/input.h [hunk 4]
```diff
@@ -271,7 +271,10 @@ class Input
     std::vector<int> out_band; // band calculation pengfei 2014-10-13
     bool out_proj_band; // projected band structure calculation jiyy add 2022-05-11
     std::vector<int> out_mat_hs; // output H matrix and S matrix in local basis.
-    bool out_mat_xc; // output exchange-correlation matrix in KS-orbital representation.
+    bool out_mat_xc; 
+    bool out_hr_npz;// output exchange-correlation matrix in KS-orbital representation.
+    bool out_dm_npz;
```
Comment (WHUweiqingzhou) [inline source/module_io/input.h:276]: please add UnitTests in `InputParaTest`

File: source/module_io/write_input.cpp [hunk 5]
```diff
@@ -228,6 +228,9 @@ ModuleBase::GlobalFunc::OUTP(ofs, "out_bandgap", out_bandgap, "if true, print ou
     ModuleBase::GlobalFunc::OUTP(ofs, "out_mat_hs2", out_mat_hs2, "output H(R) and S(R) matrix");
     ModuleBase::GlobalFunc::OUTP(ofs, "out_mat_dh", out_mat_dh, "output of derivative of H(R) matrix");
     ModuleBase::GlobalFunc::OUTP(ofs, "out_mat_xc", out_mat_xc, "output exchange-correlation matrix in KS-orbital representation");
+    ModuleBase::GlobalFunc::OUTP(ofs, "out_hr_npz", out_hr_npz, "output hr(I0,JR) submatrices in npz format");
+    ModuleBase::GlobalFunc::OUTP(ofs, "out_dm_npz", out_dm_npz, "output dmr(I0,JR) submatrices in npz format");
+    ModuleBase::GlobalFunc::OUTP(ofs, "dm_to_rho", dm_to_rho, "reads dmr in npz format and prints electron density in cube format");
```
Comment (WHUweiqingzhou) [inline source/module_io/write_input.cpp:233]: please add UnitTests in `write_input_test`

File: source/module_io/input.h (hunk 2 above)
Comment (wenfei-li) [reply source/module_io/input.h:277]: this feature is for internal use only; I am not inclined to add a detailed documentation such as to give the impression that it is also intended for external users; but descriptions are provided for the new input variables. Let me know if more is required.

**Comment by wenfei-li** [comment]
see #4066

**Pull request closed without merging**